db/controllers/users.controller.js

```diff
@@ -0,0 +1,37 @@
+const axios = require('axios');
+
+const findUser =  async (req, res) => {
+    const { email } = req.body;
+    // First, get a list of all users in the slack workspace
+    try {
+      const result = await axios.request({
+        url: 'https://slack.com/api/users.list',
+        method: 'GET',
+        headers: {
+          'Authorization': `Bearer ${process.env.SLACK_BOT_TOKEN}`
+        }
+      });
+  
+      // If the initial request is successful, look through the returned users for one with a matching email
+      if (result.data && result.data.members) {
+        const foundUser = result.data.members.find((user) => {
+          if (user.profile && user.profile.email) {
+            return user.profile.email === email;
+          }
+        });
+        if (foundUser) {
+          res.json(foundUser);
+        } else {
+          res.status(404).json({ result: 'User not found in workspace'});
+        }
+      } else {
+        throw new Error("No members present in response");
```

Comment:
We probably want some sort of response to the client. As simple as "Failed to lookup user" or some language that implies a 500-type error (i.e. don't try to re-enter your email, it's us not you). I don't know if a response is returned with throw new Error; or what it would be. :thinking:

Reply:
Since this Error is being thrown within the `try/catch`, throwing it here triggers the `catch` function. I added a 500 status to the response that gets sent in the `catch` to be more clear.